File: test/kaalkaNTP.test.js
jest.setTimeout(60000);

const KaalkaNTP = require('../kaalkaNTP');

describe('KaalkaNTP', () => {
  let kaalkaNTP;
  const timestamp = "10:15:30"; // Use a fixed, valid timestamp

  beforeEach(() => {
    kaalkaNTP = new KaalkaNTP();
  });

  test('encrypt and decrypt should work correctly', async () => {
    const originalData = 'Hello, Kaalka NTP!';
    const encryptedData = await kaalkaNTP.encrypt(originalData, timestamp);
    const decryptedData = await kaalkaNTP.decrypt(encryptedData, timestamp);

    expect(encryptedData).not.toBe(originalData);
    expect(decryptedData).toBe(originalData);
  });

  test('encrypt and decrypt should handle empty data', async () => {
    const originalData = '';
    const encryptedData = await kaalkaNTP.encrypt(originalData, timestamp);
    const decryptedData = await kaalkaNTP.decrypt(encryptedData, timestamp);

    expect(encryptedData).toBe(originalData);
    expect(decryptedData).toBe(originalData);
  });
});

File: test/kaalka_ntp_test.dart
import 'package:test/test.dart';
import 'package:kaalka/kaalka_ntp.dart';

void main() {
  group('KaalkaNTP', () {
    test('Encrypt/Decrypt with NTP (stub)', () async {
      final msg = 'NTP test';
      final encrypted = await KaalkaNTP.encryptWithNtp(msg);
      final decrypted = await KaalkaNTP.decryptWithNtp(encrypted);
      expect(decrypted, equals(msg));
    });
    test('Encrypt/Decrypt with custom time', () async {
      final msg = 'Custom NTP';
      final encrypted = await KaalkaNTP.encryptWithNtp(msg, timeKey: '01:02:03');
      final decrypted = await KaalkaNTP.decryptWithNtp(encrypted, timeKey: '01:02:03');
      expect(decrypted, equals(msg));
    });
  });
}
